codec-http2/src/main/java/io/netty/handler/codec/http2/internal/hpack/StaticTable.java

```diff
@@ -160,16 +164,16 @@ static int getIndex(byte[] name, byte[] value) {
         return -1;
     }
 
-    // create a map of header name to index value to allow quick lookup
-    private static Map<String, Integer> createMap() {
+    // create a map CharSequenceMap header name to index value to allow quick lookup
+    private static CharSequenceMap<Integer> createMap() {
         int length = STATIC_TABLE.size();
-        HashMap<String, Integer> ret = new HashMap<String, Integer>(length);
+        CharSequenceMap<Integer> ret = new CharSequenceMap<Integer>(true);
```

Comment:
Since there is only a single instance of this map ... we should consider increasing the size of the underlying array structure [1] (the max size of 128 should be fine) to reduce the probability of collisions and increase lookup speed. You will have to add a new constructor to `CharSequenceMap`.

[1] https://github.com/netty/netty/blob/4.1/codec/src/main/java/io/netty/handler/codec/DefaultHeaders.java#L119